### PR TITLE
[MFTF] Media Gallery grid attributes tests

### LINKED ISSUE #708: [MFTF] User views limited image information from the image grid in Media Gallery
# Description
Implement MFTF test to cover scenario: "User views limited image information from the image grid in Media Gallery" (story 41, #460)

#  Preconditions

1. Admin User is logged into Admin Panel
2. Adobe Stock Preview image exists in Media Gallery
3. Adobe Stock Licensed image exists in Media Gallery
4. User is signed into Adobe IMS

# Test steps

1. Open Media Gallery
2. User sees the following image attributes associated with all images:
  1. title - Asset's title
  2. type - jpg, gif, png ...
  3. dimensions
3. User doesn't see the `Unlicensed` label on licensed Adobe Stock image in Media Gallery
4. User sees the AdobeStock identification with `Unlicensed` on previewed Adobe Stock image in Media Gallery

# Additional details

Three tests should be created:
One in MediaGalleryUi module checking the attributes for uploaded image
Second in AdobeStockImageAdminUi module checking the label for Preview Adobe Stock images
Third in AdobeStockImageAdminUi module checking the label for Licensed Adobe Stock images

### PR DESCRIPTION
**Pull request by gabrieldagama**
### Description (*)

Added tests to cover the requirements on issue #708

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#708: User views limited image information from the image grid in Media Gallery

### Manual testing scenarios (*)

1. Run tests:

vendor/bin/mftf run:test AdminAdobeStockVerifyLicensedLabelTest --remove
vendor/bin/mftf run:test AdminAdobeStockVerifyUnlicensedLabelTest --remove
vendor/bin/mftf run:test AdminMediaGalleryVerifyImageGridAttributesTest --remove

### CODE REVIEW

File: AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<suites xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Suite/etc/suiteSchema.xsd">
+    <suite name="AdobeStockIntegrationEnhancedMediaGallerySuite">
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml]: Let's use `AdobeStockIntegrationSuite` introduced in https://github.com/magento/adobe-stock-integration/pull/1329

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminAdobeStockVerifyLicensedLabelTest">
+        <annotations>
+            <features value="AdobeStockImagePanel"/>
+            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/708"/>
+            <title value="User saves licensed image and verify there is no unlicensed label"/>
+            <testCaseId value="to-be-added"/>
+            <description value="Admin should be able to save licensed image and not see unlicensed label"/>
+            <severity value="CRITICAL"/>
+            <group value="adobe_stock_integration_ims_signed"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]:
```suggestion
            <group value="adobe_stock_media_gallery"/>
```

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminAdobeStockVerifyLicensedLabelTest">
+        <annotations>
+            <features value="AdobeStockImagePanel"/>
+            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/708"/>
+            <title value="User saves licensed image and verify there is no unlicensed label"/>
+            <testCaseId value="to-be-added"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]:
```suggestion
            <testCaseId value="https://studio.cucumber.io/projects/131313/test-plan/folders/1054245/scenarios/3839218"/>
```

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminAdobeStockVerifyLicensedLabelTest">
+        <annotations>
+            <features value="AdobeStockImagePanel"/>
+            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/708"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]:
```suggestion
            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/460"/>
```

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminAdobeStockVerifyLicensedLabelTest">
+        <annotations>
+            <features value="AdobeStockImagePanel"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]:
```suggestion
            <features value="MediaGallery"/>
```

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/708"/>
+            <title value="User saves licensed image and verify there is no unlicensed label"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml:17]:
```suggestion
            <title value="User views image attribution on the media gallery grid"/>
```

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+            <testCaseId value="to-be-added"/>
+            <description value="Admin should be able to save licensed image and not see unlicensed label"/>
+            <severity value="CRITICAL"/>
+            <group value="adobe_stock_integration_ims_signed"/>
+        </annotations>
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <actionGroup ref="AdminMediaGalleryEnhancedEnableActionGroup" stepKey="enableEnhancedMediaGallery">
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]: The enhanced media gallery was already enabled in the test suite precondition. This test case precondition is not needed

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+            </actionGroup>
+            <actionGroup ref="AdminOpenMediaGalleryForPageNoEditorActionGroup" stepKey="openMediaGalleryForPage"/>
+            <actionGroup ref="AdminAdobeStockOpenFromEnhancedMediaGalleryActionGroup" stepKey="openAdobeStockPanel"/>
+        </before>
+        <after>
+            <actionGroup ref="AdminEnhancedMediaGalleryImageDeleteActionGroup" stepKey="removeSavedPreview"/>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logout"/>
+        </after>
+
+        <actionGroup ref="AdminSearchImagesOnModalActionGroup" stepKey="searchForNotPreviewedLicensedImageToSave">
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]: Let's make test step names a bit more readable
```suggestion
        <actionGroup ref="AdminSearchImagesOnModalActionGroup" stepKey="searchForLicensedImageToSave">
```

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyUnlicensedLabelTest.xml [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminAdobeStockVerifyUnlicensedLabelTest">
+        <annotations>
+            <features value="AdobeStockImagePanel"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyUnlicensedLabelTest.xml]: Please apply the same changes to the annotation as mentioned in the previous test.

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+            <argument name="query" value="{{AdobeStockLicensedImage.id}}"/>
+        </actionGroup>
+        <actionGroup ref="AdminAdobeStockExpandImagePreviewActionGroup" stepKey="expandLicensedImage"/>
+        <actionGroup ref="AdminAdobeStockSaveLicensedActionGroup" stepKey="saveLicensedImage"/>
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml:37]: To download a licensed image the user must be logged in to Adobe Stock, I cannot find the login step in the test.

File: MediaGalleryUi/Test/Mftf/Test/AdminMediaGalleryVerifyImageGridAttributesTest.xml [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminMediaGalleryVerifyImageGridAttributesTest">
+        <annotations>
+            <features value="MediaGallery"/>
+            <useCaseId value="https://github.com/magento/adobe-stock-integration/issues/708"/>
+            <title value="Verify image grid attributes"/>
+            <testCaseId value="to-be-added"/>
```
Comment (sivaschenko) [inline MediaGalleryUi/Test/Mftf/Test/AdminMediaGalleryVerifyImageGridAttributesTest.xml:15]: Please reference the same story and test case as for other tests (I believe there is currently only one test case that checks attributes and labels)

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @gabrieldagama ! Please see my review comments

File: AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml (hunk 1 above)
Comment (gabrieldagama) [reply AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml]: Hi @sivaschenko, the AdobeStockIntegrationSuite disables the EnhancedMediaGallery as shown below:
`            <actionGroup ref="AdminMediaGalleryEnhancedEnableActionGroup" stepKey="disableEnhancedMediaGallery">
                <argument name="enabled" value="0"/>
            </actionGroup>`

Should we use AdobeStockIntegrationSuite and for each test that the Enhanced Media Gallery is relevant we will have to enable it on the before node?

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml (hunk 2 above)
Comment (gabrieldagama) [reply AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]: In order to verify a Licensed label we need to save a Licensed image, to do so, we need to be logged in. By changing this group we will have to perform all the login steps within the <before> node in the tests, is that the suggestion?

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml (hunk 7 above)
Comment (gabrieldagama) [reply AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml]: Looking at the AdobeStockIntegrationImsSignedSuite we have the following:
`            <actionGroup ref="AdminMediaGalleryEnhancedEnableActionGroup" stepKey="disableEnhancedMediaGallery"/>
`
Which is disabling the enhanced media gallery, if we change this to the one suggested above we will have to perform the login steps on the <before> node.

File: AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml (hunk 10 above)
Comment (gabrieldagama) [reply AdobeStockImageAdminUi/Test/Mftf/Test/AdminAdobeStockVerifyLicensedLabelTest.xml:37]: That is being done through the AdobeStockIntegrationImsSignedSuite.

File: AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml (hunk 1 above)
Comment (sivaschenko) [reply AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml]: Sorry, corrected the name in my PR to `AdobeStockMediaGallerySuite`

File: AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <actionGroup ref="AdminDisableWYSIWYGActionGroup" stepKey="disableWYSIWYG" />
+            <actionGroup ref="AdminAdobeStockSetConfigActionGroup" stepKey="setCorrectModuleConfig"/>
+            <actionGroup ref="AdminMediaGalleryEnhancedEnableActionGroup" stepKey="enableEnhancedMediaGallery">
+                <argument name="enabled" value="1"/>
+            </actionGroup>
+        </before>
+        <after>
+            <actionGroup ref="AdminEnableWYSIWYGActionGroup" stepKey="enableWYSIWYG" />
```
Comment (Nazar65) [inline AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml]: I think we need to add AdminLogout action in after block

**Review by Nazar65 — CHANGES_REQUESTED** [review]

File: AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml (hunk 12 above)
Comment (gabrieldagama) [reply AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml]: Hi @Nazar65, none of the others Suites implement an admin logout on after block, I believe each test starts with a clear session, but this Suite was deleted anyway, thanks for your comment.

Comment (Nazar65) [reply AdobeStockImageAdminUi/Test/Mftf/Suite/AdobeStockIntegrationEnhancedMediaGallerySuite.xml]: > To wrap up: The lack of adminLogoutActionGroup in after section does not affect other tests, however it's considered to be best practice, as (!) you should follow the scout rule - leave the place better than you found when you started.

**Comment by gabrieldagama** [comment]
@magento run Functional Tests CE

File: AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockLicensedImageActionGroup.xml [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminSaveAdobeStockLicensedImageActionGroup">
+        <waitForElementVisible selector="{{AdminAdobeStockSaveImagePreviewSection.name}}" time="30" stepKey="waitForInputVisible"/>
+        <click selector="{{AdminAdobeStockImagePreviewSection.confirm}}" stepKey="confirmSavingLicensed"/>
+        <waitForPageLoad stepKey="waitForImageToSave"/>
```
Comment (Nazar65) [inline AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockLicensedImageActionGroup.xml]: It would be better to waitForLoadingMaskToDisappear

**Review by Nazar65 — CHANGES_REQUESTED** [review]

File: AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockLicensedImageActionGroup.xml (hunk 13 above)
Comment (gabrieldagama) [reply AdobeStockImageAdminUi/Test/Mftf/ActionGroup/AdminSaveAdobeStockLicensedImageActionGroup.xml]: Well spotted, thanks @Nazar65

**Comment by sivaschenko** [comment]
@magento run Functional Tests CE,Functional Tests EE,Functional Tests B2B

**Comment by sivaschenko** [comment]
@magento run Functional Tests CE,Functional Tests EE,Functional Tests B2B

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**